cypress/e2e/settings/theme-choice-panel.spec.ts

```diff
@@ -0,0 +1,128 @@
+
+                cy.get(".mx_ThemeSelectors").within(() => {
+                    cy.get(".mx_ThemeSelector_light").should("exist");
+                    cy.get(".mx_ThemeSelector_dark").should("exist");
+
+                    // Assert that the light theme is selected
+                    cy.get(".mx_ThemeSelector_light.mx_StyledRadioButton_enabled").should("exist");
+
+                    // Assert that disabled buttons should not exist
+                    cy.get(".mx_StyledRadioButton_disabled").should("not.exist");
+                });
+
+                // Assert that the checkbox for the high contrast theme is rendered
+                cy.contains(".mx_Checkbox", "Use high contrast").should("exist");
+            });
+    });
+
+    it(
+        "should disable the labels for themes and the checkbox for the high contrast theme if the checkbox for " +
+            "the system theme is clicked",
+        () => {
+            cy.openUserSettings("Appearance")
+                .get(".mx_ThemeChoicePanel")
+                .get("[data-testid='checkbox-use-system-theme'] .mx_Checkbox")
+                .click();
+
+            cy.get(".mx_ThemeChoicePanel").within(() => {
+                // Assert that the labels for the light theme and dark theme are disabled
+                cy.get("label.mx_StyledRadioButton_disabled").should("have.length", 2);
+
+                // Assert that there does not exist a label for an enabled theme
+                cy.get("label.mx_StyledRadioButton_enabled").should("not.exist");
+
+                // Assert that the checkbox and label to enable the the high contrast theme should not exist
+                cy.get("[data-testid='checkbox-use-high-contrast']").should("not.exist");
+            });
+        },
+    );
+
+    it("should not render the checkbox and the label for the high contrast theme if the dark theme is selected", () => {
+        cy.openUserSettings("Appearance");
+
+        // Assert that the checkbox and the label to enable the high contrast theme should exist
+        cy.get("[data-testid='checkbox-use-high-contrast']").should("exist");
+
+        // Enable the dark theme
+        cy.get(".mx_ThemeSelector_dark").click();
+
+        // Assert that the checkbox and the label should not exist
+        cy.get("[data-testid='checkbox-use-high-contrast']").should("not.exist");
+    });
+
+    it("should support enabling the high contast theme", () => {
+        cy.createRoom({ name: "Test Room" }).viewRoomByName("Test Room");
+
+        cy.get(".mx_GenericEventListSummary").within(() => {
+            // Assert that $primary-content is applied to GELS summary on the light theme
+            // $primary-content on the light theme = #17191c = rgb(23, 25, 28)
+            cy.get(".mx_TextualEvent.mx_GenericEventListSummary_summary")
+                .should("have.css", "color", "rgb(23, 25, 28)")
+                .should("have.css", "opacity", "0.5");
```

Comment:
Please note that this opacity value makes `TexualEvent` color on the light theme grey.